### PR TITLE
🏗Always build amp-loader and amp-auto-lightbox.

### PR DESCRIPTION
**Pull request by sparhami**
It is easy to forget to build these extensions when running locally.
Since these are loaded by the runtime, they should be built in order to
get a more accurate representation of how AMP behaves when manually
testing changes.

- Remove unnecessary `@private` for things that are not exported.
- Remove unnecessary type that can be inferred.

### CODE REVIEW

File: build-system/tasks/extension-helpers.js [hunk 1]
```diff
@@ -206,6 +209,8 @@ function getExtensionsToBuild() {
     extensionsToBuild = dedupe(extensionsToBuild.concat(extensionsFrom));
   }
 
+  extensionsToBuild = dedupe(extensionsToBuild.concat(DEFAULT_EXTENSION_SET));
```
Comment (rsimha) [inline build-system/tasks/extension-helpers.js]: This function will read better if you initialize `extensionsToBuild` with `DEFAULT_EXTENSION_SET` on line 196 and then change line 204 to `extensionsToBuild = dedupe(extensionsToBuild.concat(argv.extensions.split(',')))`.

**Comment by sparhami** [comment]
Had to make some more changes, since there was an assumption that having no extensions to build meant to build all the extensions. I fixed dep-check which was relying on this assumption, but I may have missed others that also have the same problem.

**Comment by sparhami** [comment]
Looks like this actually also applies to the dist target. I think I need to change the approach here.

**Comment by rsimha** [comment]
> Looks like this actually also applies to the dist target. I think I need to change the approach here.

You're right. There's another problem too. `buildExtensions()` in `build-system/tasks/extension-helpers.js` does nothing when `--noextensions` is passed in to `gulp`. This means `DEFAULT_EXTENSION_SET` won't actually get built.

Let me play around

**Comment by sparhami** [comment]
> I'll try to come up with a working version that includes your changes until [454560e](https://github.com/ampproject/amphtml/commit/454560e2780b78aa55e346ac7e911e525124aad2). SG?

Sounds good.

**Comment by rsimha** [comment]
@sparhami I just sent you #24193 for review. It should effectively replace this PR. Feel free to rest locally and let me know what you think.

**Pull request closed without merging**